### DISCUSSION TITLE
How to enable a formatter installed via Mason.nvim?

### DISCUSSION THREAD

**Discussion by V4G4X** [mason-org/mason.nvim#1625] [Q&A] [answered]
I had hoped this would work out of the box but here goes.  

My goal is to use [gofumpt](https://github.com/mvdan/gofumpt) as my Golang formatter.  

1. I installed gofumpt using `go install mvdan.cc/gofumpt@latest`. 
2. I installed `gofumpt` using Mason.
3. But when I format my file, it does the basic `gofmt` formatting and not `gofumpt`.  
4. If I add the following config to my lua initialisation:
```lua
require('lspconfig').gopls.setup({
    settings = {
        gopls = {
            gofumpt = true
        }
    }
})
```
Then it works, but I thought the point of Mason was to avoid this?

**Comment by WhoIsSethDaniel** [comment, marked as answer]
If you want to use the externally installed (via mason) `gofumpt`  you will want to have a plugin that deals with the formatting setup for you. f.e. [conform.nvim](https://github.com/stevearc/conform.nvim).

In your 4) you turn on `gofumpt` formatting via `gopls`. This *does not* use  an externally installed `gofumpt` (or `gofmt`). Instead it uses the `gofumpt` library and compiles the `gofumpt` (or `gofmt`) functionality into `gopls`. 

What `mason.nvim` does is make it easy to download third-party tools, such as formatters, languages servers, linters, etc.... 

If you already use `gopls` I'd recommend using `mason` to install `gopls` and configure it to format as you did in 4). You can then remove installing `gofumpt` via mason since it won't be required. `mason` does not configure `gopls` for you ([mason-lspconfig](https://github.com/williamboman/mason-lspconfig.nvim) also does not configure `gopls` for you, but it can make *when* to configure it simpler).

**Reply by V4G4X** [reply]
So,  

- Mason only installs LSP servers, Formatters and similar tools. 
- LspConfig only connects those installed LSP servers, to my LSP Client(nvim)

Am I right so far?

**Reply by WhoIsSethDaniel** [reply]
`nvim-lspconfig` also gives you a bare-minimum configuration for running language servers. In plenty of cases this means no actual configuration since the language server will run fine with built-in defaults. e.g. `gopls` runs fine with no provided configuration. (If you look at the `gopls` [config in the nvim-lspconfig repo](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/gopls.lua) everything in that file is for setting up neovim and how to launch `gopls`.)

Anyway, I think your two bullet-points are correct.
